plugins/octopus-deploy/src/api/index.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  createApiRef,
+  DiscoveryApi,
+  IdentityApi,
+} from '@backstage/core-plugin-api';
+
+/** @public */
+export type OctopusProgression = {
+  Environments: OctopusEnvironment[];
+  Releases: OctopusReleaseProgression[];
+};
+
+/** @public */
+export type OctopusEnvironment = {
+  Id: string;
+  Name: string;
+};
+
+/** @public */
+export type OctopusReleaseProgression = {
+  Release: OctopusRelease;
+  Deployments: { [key: string]: OctopusDeployment[] };
+};
+
+/** @public */
+export type OctopusRelease = {
+  Id: string;
+  Version: string;
+};
+
+/** @public */
+export type OctopusDeployment = {
+  State: string;
+};
+
+/** @public */
+export const octopusDeployApiRef = createApiRef<OctopusDeployApi>({
+  id: 'plugin.octopusdeploy.service',
+});
+
+const DEFAULT_PROXY_PATH_BASE = '/octopus-deploy';
+
+/** @public */
+export type Options = {
+  discoveryApi: DiscoveryApi;
+  identityApi: IdentityApi;
+  /**
+   * Path to use for requests via the proxy, defaults to /octopus-deploy
+   */
+  proxyPathBase?: string;
+};
+
+/** @public */
+export interface OctopusDeployApi {
+  getReleaseProgression(
+    projectId: string,
+    releaseHistoryCount: number,
+  ): Promise<OctopusProgression>;
+}
+
+/** @public */
+export class OctopusDeployClient implements OctopusDeployApi {
+  private readonly discoveryApi: DiscoveryApi;
+  private readonly identityApi: IdentityApi;
+  private readonly proxyPathBase: string;
+
+  constructor(options: Options) {
+    this.discoveryApi = options.discoveryApi;
+    this.identityApi = options.identityApi;
```

Comment:
Could you inject the `fetchApiRef` instead of the `identityApi`? It's pretty much exactly `fetch`, BUT it is easy to replace, it can have metrics and tracking etc on it, and also - it does inject the backstage token onto requests to the backend without you having to think about it.

Reply:
I didn't know about that, but that does make things a little bit easier. Thanks for the suggestion.